app/assets/javascripts/git.js.erb

```diff
@@ -32,10 +32,24 @@ $j(document).ready(function () {
 
                     if (tree.is_closed(node) && node.data.childAnnotations && node.data.childAnnotations.length) {
                         element.addClass('git-annotated-parent');
+                        var groups = {};
                         node.data.childAnnotations.forEach(function(annotation) {
-                            var template = HandlebarsTemplates["git/annotation"];
-                            element.append(template(annotation));
+                            if (!groups[annotation.key]) {
+                                // Don't include `count` when only one single annotation.
+                                groups[annotation.key] = { key: annotation.key, label: annotation.label };
+                            } else {
+                                if (!groups[annotation.key].count) {
+                                  groups[annotation.key].count = 1;
+                                }
+                                groups[annotation.key].count++;
+                            }
                         });
+                        for (var key in groups) {
+                            if (groups.hasOwnProperty(key)) {
+                                var template = HandlebarsTemplates["git/annotation"];
+                                element.append(template(groups[key]));
+                            }
+                        }
```

Comment:
pure pedantry, and only because I've just been doing this with the Nels javascript - but some of these `var` definitions could be `const` or `let`
Not a blocker though

Reply:
Yeah I just kept it consistent with the other code. We should do an ES6 refactor at some point.